index.js

```diff
@@ -12,7 +12,58 @@ function setMultipart (req, done) {
   done()
 }
 
+function attachToBody (options, req, reply, next) {
+  req[kMultipart] = true
+
+  const defaultConsumer = (field, file, filename, encoding, mimetype) => {
+    const fileData = []
+    file.on('data', data => { fileData.push(data) })
+    file.on('limit', () => { body[field].limit = true })
+    file.on('end', () => {
+      body[field].data = Buffer.concat(fileData)
+    })
+  }
+
+  const consumerStream = options.onFile || defaultConsumer
+  const body = { }
+  const mp = req.multipart((field, file, filename, encoding, mimetype) => {
+    body[field] = {
+      data: [],
+      filename,
+      encoding,
+      mimetype,
+      limit: false
+    }
+    consumerStream(field, file, filename, encoding, mimetype)
+  }, function (err) {
+    req.body = body
```

Comment:
Is this correct? What’s your reasoning?

Reply:
After composing the `body` object I will attach it to the request body.
This is the function executed on `done` that could and could not have the error.
Should I check the error before adding the body to the request?

Reply:
I think so, yes.